model/host/db.go

```diff
@@ -483,3 +484,54 @@ func inactiveHostCountPipeline() []bson.M {
 		},
 	}
 }
+
+// struct for storing Id and FinishTime pairs
+type FinishTime struct {
+	Id         string    `bson:"_id"`
+	FinishTime time.Time `bson:"finish_time"`
+}
+
+// aggregation pipeline to compute latest finish time for running hosts with child containers
+func lastContainerFinishTimePipeline() []bson.M {
+	return []bson.M{
+		{
+			// matches all containers
+			"$match": bson.M{
+				ParentIDKey: bson.M{"$exists": true},
+			},
+		},
+		{
+			// joins hosts and tasks collections on task ID
+			"$lookup": bson.M{
+				"from":         "tasks",
+				"localField":   RunningTaskKey,
+				"foreignField": IdKey,
+				"as":           "task",
+			},
+		},
+		{
+			// deconstructs $lookup array
+			"$unwind": "$task",
+		},
+		{
+			// groups containers by parent host ID
+			"$group": bson.M{
+				"_id": "$" + ParentIDKey,
+				"finish_time": bson.M{
+					// computes last container finish time for each host
+					"$max": bson.M{
+						"$add": []interface{}{"$task.start_time",
+							bson.M{"$divide": []interface{}{"$task.duration_prediction.value", time.Millisecond}},
```

Comment:
I think 1000000 is more appropriate here than time.Millisecond because you aren't trying to divide a time by another time. I would also add a comment about why this is necessary